### DISCUSSION TITLE
List of Questions to Ask Client

### DISCUSSION THREAD

**Discussion by wqyeo** [Juicy-Lemonberry/INF2001-P5-2#13] [Ideas]
**Chip in ideas on what you would like to ask the client on here.** 

During a latter date _(near meetings with client)_, we will collect all questions, and sort/categorize them by priority before using it as a question list towards the Client meetings.

- [ ] What type of companies are we dealing with?
- [ ] Is the created system based on site?
- [ ] ...
- [ ] 
> _Please add them in as comments, do not edit the post directly_

**Comment by wqyeo** [comment]
### How does the management hierarchy look like?

The current assumption is that, Chiefs will manage site leaders' workload, while site leaders will manage staff's workload.

----

### Who will be adding in the information of the site/area data? _(aka, workload data)_

Currently, no information is given on how the application will obtain workload data.

> Also, might want to ask how many sites/areas are there.

-----

### How to determine the starting point of an area?

Currently, we know that an area consists of 2 or more sites within 10km radius; But, how does one define the starting point of an area?

**Reply by wqyeo** [reply]
- Are there any other stakeholders who would interact with the system?

**Comment by 2202348** [comment]
- How should employees and managers be notified about new job assignments, changes to assignments, or rejected assignments?
- What are the relevant information needed to make the job assignment easier on the job allocation page?
- What information should be displayed to the staff when they view their weekly job assignments and overall workload for the month? (site, area, timing etc)
- After a staff discusses the job rejection with their manager, will the manager be the one who manually removes the job assigned to them or will the staff remove it themselves?

**Comment by Zhwee** [comment]
-how does client want it(the management web) to look like

**_-Has he used other management apps before? why did he stop using them? what were the apps lacking, or what was good(features)?
-if he say never use other management apps before, then be prepared with examples already, show them the management apps features, ask him what he likes and don't like_**


> Staff should be able to view their weekly job assignments and overall workload for the month
on their landing page
The manager should be able to visualise the staff workload immediately on the landing page
-so these parts, for indiv employee, does he want to see it in list format, calendar format, or both (able to toggle)?
-what about manager, does he want to see it in calendar format?

> Individual employee workload and availabilities should be able to be visualised at a glance.
-so what are some of the main things he want to see at the main page? give him examples, hope he can be specific about it

-does client want the sites to be seperated? diff sites, diff groups that can register to (so, only those leaders within the site are authorized to see specific items ==kinda like slack)

> The company classifies tenured sites into small and large sizes and number of manned gates.
-so does he want an on/off shift kind of thing as well, like who is on shift, and who is off duty (does he also want people to login here, so if they are on shift, can be updated here automatically, if they submit medical cert, it automatically shows red i.e. whereever the dude was supposed to be at, someone else now needs to fill up that slot)

-do we also need a page for cars, i.e. there should be a limited number of company cars, and not everyone travel at the same time, so when it should be deployed etc?

-how many company cars are there?

-does he want a leaderboard page lmao, like employee of the month kinda stuff (see who work the most no. of hours)

> The manager should be able to view up to three staff availability and any relevant information
to make the job assignment easier on the job allocation page
-up to 3? why not more, unless this means there's only 3 people always on standby?
-if so, that means we need to add this feature into the prototype i.e. employees have the main work timings, but they are also placed on backup (for either they want to do OT//tho need to take note SG max weekly working hour is 44h, so anytime after that is considered OT)

-for OT hours, do we separate them out? or just add it into the employees profile (i.e. total no. of hours worked this week)

-does the client also want the total cost/budget/salary of the worker to be calculated for them in the app? (like show base pay, and extra OT pay)

**_-What if we think of all the diff pages that should be in the webpage alr, then ask him, afterwards, what other page would he want/don't want to add based on what we've done (ngl i don't see a prob if we drew it out, then ask about what else he want or don't want==i.e. diff pov diff visualisation)_**

> Staff can indicate their job preference for the week
-so every week will have message prompting the employee which job he wants to do?
**_-what types of jobs are there then? what are the diff job preferences that can be chosen?_**
-what type of jobs are assigned? can give examples? be specific and general/broad

-how robust does he want the login page to be, like login using work email? like slack?

**Reply by Zhwee** [reply]
-how do you want the updates to look like

-what about notifications, any pop-ups needed? when would pop-ups be needed

-does he want it to be easy to use, so easy that nobody needs to teach them anything, and they know how to use already?
-do you want us to include pop-up instructions when you first enter the website, the basic runthrough of how the system works

-do the shifts include toilet break, meal break? if have, how long are the breaks?

-how to assign the job from employers/leaders, like kanban liddat, just click (+) at the side, then type in the name to find the person, then it assign job to that person?
-then that person gets a notif about the new job assigned

**Reply by Zhwee** [reply]
![image](https://github.com/Juicy-Lemonberry/INF2001-P5-2/assets/77541619/bfbcb913-b17c-4818-85af-ddb0b3af6446)

**Comment by ivriahtyx** [comment]
- what needs to be done to the system if someone misses the weekly deadline for job allocation and staff availability? Does the system just assumes that they are not available for the week?

- What are the informations that the employee would like to see at a glance at the loading page?
 
- Staff are allowed to reject jobs assigned to them but will be warned to discuss job with their manager before proceeding with the rejection. Does this means that it will be in pending state until manager approves it? Should the system send email or some sort of notification to the manager automatically upon staff rejection of jobs?

**Comment by 2202282** [comment]
- How does communication flow between different levels of the hierarchy for workload-related matters?
- How is the accuracy of the information maintained, especially for site and area data that might change over time?
- How do you envision the process of job assignment? Should it be a manual process, or do you prefer automation based on availability and preferences?
- Are there any specific criteria or constraints that should be considered during job assignment? (e.g., skills required, preferred shifts, proximity to site)
- Should there be a time limit or deadline for the manager's decision after a job rejection is discussed?
- Would you like to include any features like leaderboards, recognition for high performers, or any gamification elements in the application?
- Are there any specific legal or regulatory requirements that the system should comply with in terms of data privacy?
- How many users are expected to use the system simultaneously? Do you have performance expectations in terms of response times?
- Is there any existing software, databases, or systems that the new workload management system should integrate with?
- Are there any specific reports or analytics that managers would like to generate from the system, such as workload distribution trends or employee performance metrics?

**Comment by wqyeo** [comment]
# Quest about the current system _(before implementation)_

- Can you describe the current process of workload management and job allocation? Are there any pain points or challenges with the existing system?

- How do employees currently communicate their availability and preferences for workload assignments?

- Are there any rules or guidelines around how staff can indicate their preferences or limitations?

- Are there any specific criteria or considerations that managers use when assigning jobs to employees?

> To understand the current workflow

**Reply by wqyeo** [reply]
Additionally, can managers manage *all* the workers? Or are a dedicated number of staff/workers assigned to each manager?
